### DISCUSSION TITLE
attachment_token in che formato?

### DISCUSSION THREAD

**Discussion by aster94** [fattureincloud/api#266] [Q&A] [answered]
Salve,

Per caricare un allegato a un acquisto come bisognerebbe procedere?
attachment_token è in base64?
Non sto usando l'SDK

Grazie

**Comment by emajo** [comment, marked as answer]
Ciao @aster94, [qui](https://developers.fattureincloud.it/docs/guides/invoice-creation#8%EF%B8%8F⃣-step-eight-attachment) è descritto il flusso per il caricamento di un allegato.

Grazie,
Emanuele.

**Reply by aster94** [reply]
okkk, ma quando non si usa l'SDK che formato dovrebbe avere il token?

**Reply by valmoz** [reply]
Buongiorno @aster94 , il token deve essere inserito esattamente così come viene restituito dal metodo [Upload Attachment](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/issued_documents/attachment).
Quindi, se l'API ritorna qualcosa del tipo:
```json
{
"data": {
"attachment_token": "YmMyNWYxYzIwMTU3N2Y4ZGE3ZjZiMzg5OWY0ODNkZDQveXl5LmRvYw"
}}
```

allora è necessario passare esattamente la stringa contenuta in attachment_token.

Riassumendo:
- Deve usare il metodo Upload Attachment per caricare l'allegato
- Dalla risposta recupera l'attachment_token
- Nella chiamata di creazione deve inserire l'attachment_token esattamente come fornito dalle API

N.B.: le SDK sono solamente dei wrapper delle API, quindi il loro comportamento è esattamente lo stesso. 

Grazie
Mauro

**Reply by aster94** [reply]
Grazie, mi era sfuggito quel metodo e infatti non stavo capendo come procedere, adesso è chiaro!
